components/ai_chat/core/common/utils.cc

```diff
@@ -0,0 +1,25 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/ai_chat/core/common/utils.h"
+
+#include "brave/brave_domains/service_domains.h"
+#include "brave/components/ai_chat/core/common/constants.h"
+#include "url/gurl.h"
+#include "url/url_constants.h"
+
+namespace ai_chat {
+
+bool IsBraveSearchURL(const GURL& url) {
+  return url.is_valid() && url.SchemeIs(url::kHttpsScheme) &&
+         url.host_piece() ==
+             brave_domains::GetServicesDomain(kBraveSearchURLPrefix);
+}
+
+bool IsOpenLeoButtonFromBraveSearchURL(const GURL& url) {
+  return IsBraveSearchURL(url) && url.path_piece() == "/leo";
```

Comment:
should we also check:
1. There is a fragment
2. The fragment is non-empty?

as I understand it from the spec, the OpenLeo url is invalid unless it has a nonce, and the nonce matches whats in the url

Reply:
yeah the nonce gets validated elsewhere but an empty one should be rejected anyways so we can check it here

Reply:
Added in 65e3da2b293ed6359d66a37e6540f90a1db58a1b